src/ImageSharp/Processing/Processors/Convolution/KernelSamplingMap.cs

```diff
@@ -105,5 +81,79 @@ public void Dispose()
                 this.isDisposed = true;
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void BuildOffsets(IMemoryOwner<int> offsets, int boundsSize, int kernelSize, int min, int max, BorderWrappingMode borderMode)
+        {
+            int radius = kernelSize >> 1;
+            Span<int> span = offsets.GetSpan();
+            ref int spanBase = ref MemoryMarshal.GetReference(span);
+            for (int chunk = 0; chunk < boundsSize; chunk++)
+            {
+                int chunkBase = chunk * kernelSize;
+                for (int i = 0; i < kernelSize; i++)
+                {
+                    Unsafe.Add(ref spanBase, chunkBase + i) = chunk + i + min - radius;
+                }
+            }
+
+            this.CorrectBorder(span, kernelSize, min, max, borderMode);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void CorrectBorder(Span<int> span, int kernelSize, int min, int max, BorderWrappingMode borderMode)
+        {
+            var affectedSize = (kernelSize >> 1) * kernelSize;
+            if (affectedSize > 0)
+            {
+                switch (borderMode)
+                {
+                    case BorderWrappingMode.Repeat:
+                        Numerics.Clamp(span.Slice(0, affectedSize), min, max);
+                        Numerics.Clamp(span.Slice(span.Length - affectedSize), min, max);
+                        break;
+                    case BorderWrappingMode.Mirror:
+                        for (int i = 0; i < affectedSize; i++)
+                        {
+                            var value = span[i];
+                            if (value < min)
+                            {
+                                span[i] = min - value + min;
```

Comment:
Not sure whether this would really affect performance as it depends on the span size but you can cache `2 * min` before the loop:
```cs
// if min is always positive (uint)min * 2 for maximum performance 
// due to binary shift multiplication
int min2 = min + min;
for (int i = 0; i < affectedSize; i++)
{
    var value = span[i];
    if (value < min)
    {
        span[i] = min2 - value;
    }
}
```

Reply:
Using `ref`'s instead of span indexing should also yield better performance.

Reply:
Added `min2` variable, as suggested.

Typical `kernelSize` values are 3 and 5. Using references does however reduce readability. 
Do you think it is worth it?

Reply:
Agreed on readability, didn't know about typical sizes so yeah, it shouldn't matter.